doc/releases/release-notes-1.14.rst

```diff
@@ -36,7 +36,59 @@ Drivers and Sensors
 Networking
 **********
 
-* TBD
+* The :ref:`BSD socket API <bsd_sockets_interface>` should be used by
+  applications for any network connectivity needs.
+* Majority of the network sample applications are converted to use
+  BSD socket API.
+* New BSD socket based APIs created for these components and protocols:
+
+  - :ref:`MQTT <mqtt_socket_interface>`
+  - :ref:`CoAP <coap_sock_interface>`
+  - :ref:`LWM2M <lwm2m_interface>`
+  - SNTP
+* net-app client and server API are removed. Following net-app based APIs are
+  removed:
+
+  - MQTT
+  - CoAP
+  - SNTP
+  - LWM2M
+  - HTTP client & server
+  - Websocket
+* Network packet (:ref:`net-pkt <net_pkt_interface>`) API overhaul. The new
+  net-pkt API uses less memory and is more streamlined than the old one.
+* Implement `getnameinfo()`, `shutdown()`, `freeaddrinfo()`, `gethostname()`,
+  `getsockopt()`, `setsockopt()`, `select()`
+* Convert BSD socket code to use global file descriptor numbers.
+* Network subsystem converted to use new :ref:`logging system <logger>`.
+* Allow disabling IPv4, IPv6, UDP and TCP simultaneously.
+* Add support for :ref:`BSD socket offloading <net_offload_interface>`.
+* Add support for long lifetime IPv6 prefixes.
+* Add support for IPv6 multicast address check enhancements.
+* Add support for IPv6 Destination Options Header extension.
+* Add packet socket (AF_PACKET) support.
+* Add Socket CAN (AF_CAN) support.
+* Add SOCKS5 proxy support to MQTT client.
+* Add IPSO Timer object support to LWM2M.
+* Add :ref:`sample application <google-iot-mqtt-sample>` for Google IoT Cloud.
+* :ref:`Network interface <net_if_interface>` numbering starts now from 1.
```

Comment:
Perhaps say "from 1 for POSIX compatibility".

Reply:
Mentioning POSIX here does not sound very relevant, we are free to number the interfaces as we like.

Reply:
I'd say the fact that "we are free to number the interfaces as we like" isn't very relevant. What's important to users is that we exert effort to be compatible with well-known standard APIs. That said, if we want to keep release notes short and cryptic, well...

Reply:
I added the posix thingy to new version.